### ISSUE TITLE
Connecting MQTT server with login and password

### ISSUE THREAD

**Issue by Louloudu79** [EnricoFerro/yi-hack-to-mqtt#3]
Hello EnricoFerro,

I really enjoy the idea of your project to be able to integrate the YI Hack camera with Home Assistant. I'm trying to install it with Docker on a RPi 3B+ with the latest Home Assistant 0.118 and Mosquitto as a broker on the same RPi.

But I'm struggling to make it works. The yi-hack-to-mqtt log says : 

```
yi-hack-2-mqtt@0.0.1 start:prod
node dist/main
[Nest] 17   - 11/24/2020, 7:44:54 PM   [NestFactory] Starting Nest application...
[Nest] 17   - 11/24/2020, 7:44:54 PM   [InstanceLoader] ClientsModule dependencies initialized +105ms
[Nest] 17   - 11/24/2020, 7:44:54 PM   [InstanceLoader] DiscoveryModule dependencies initialized +4ms
[Nest] 17   - 11/24/2020, 7:44:54 PM   [InstanceLoader] ConfigHostModule dependencies initialized +1ms
[Nest] 17   - 11/24/2020, 7:44:54 PM   [InstanceLoader] ScheduleModule dependencies initialized +3ms
[Nest] 17   - 11/24/2020, 7:44:54 PM   [InstanceLoader] ConfigModule dependencies initialized +2ms
[Nest] 17   - 11/24/2020, 7:44:54 PM   [InstanceLoader] YiCamModule dependencies initialized +4ms
[Nest] 17   - 11/24/2020, 7:44:54 PM   [NestMicroservice] Nest microservice successfully started +46ms
[Nest] 17   - 11/24/2020, 7:44:54 PM   [Server] Object:
{
  "code": 5
}
 +229ms
[Nest] 17   - 11/24/2020, 7:44:55 PM   [Server] Object:
{
  "code": 5
}
```

It repeats the code 5 error until I stop the container.

I also check my mosquitto server log, the connexion to the MQTT server seems to be blocked. It logs this at each time the Code 5 error is generated : 

```
1606083310: New connection from "IP.OF.DOCKER.CONTAINER" on port "MQTTSERVERPORT".
1606083310: Socket error on client <unknown>, disconnecting.
```

So my trouble seems to come from the connexion to the Mosquitto server. I'm wondering if the fact that my Mosquitto server is protected by a login and password is an issue? Because it is not possible to add these connexion parameters to the configuration file.

Thank you in advance for your help and your work,

Louloudu79

**Comment by EnricoFerro** [comment]
Hi Louloudu79,

I've updated the libraries  of mqtt and nestjs  and make some test with the mqtt login.
A new version is released 0.0.2.

I've added in the documentation the syntax `server: mqtt://admin:password@hostname:port` and with this syntax is possible to connect to mqtt with login and password. 

Enrico

**Comment by Louloudu79** [comment]
Hi EnricoFerro,

I installed it again after having changed the parameters according to the new options and it has solved my troubles. Everything is working as expected! 

Thank you very much for this update!

Louloudu79
